cyral/resource_cyral_repository.go

```diff
@@ -33,35 +33,38 @@ type RepoData struct {
 	Properties          *RepositoryProperties `json:"properties,omitempty"`
 }
 
-func (data *RepoData) IsReplicaSet() bool {
-	return data.Properties != nil && data.Properties.MongoDBServerType == mongodbReplicaSetServerType
-}
-
 func (data *RepoData) WriteToSchema(d *schema.ResourceData) {
 	d.Set("type", data.RepoType)
 	d.Set("host", data.Host)
 	d.Set("port", data.Port)
 	d.Set("name", data.Name)
 	d.Set("labels", data.Labels)
 
-	var properties []interface{}
-	if data.Properties != nil {
-		propertiesMap := make(map[string]interface{})
+	properties := data.PropertiesAsInterface()
+	d.Set("properties", properties)
+}
 
+func (data *RepoData) PropertiesAsInterface() []interface{} {
+	properties := []interface{}{}
```

Comment:
I think this is more idiomatic and it's also simpler than the `DiffSuppressFunc` logic. Please change to that instead and remove the `DiffSuppressFunc`, thanks.
```suggestion
	if properties := data.PropertiesAsInterface(); properties != nil {
		d.Set("properties", properties)
	}
}

func (data *RepoData) PropertiesAsInterface() []interface{} {
	var properties []interface{}
```

Reply:
This works! Thanks Victor!

I don't know why I didn't try this. Let me at least explain why I went through so much trouble, maybe it will help you in the future too, to not get confused as I did. If you type in `properties {}` or NOTHING in your Terraform config, you will get the same value from `d.Get("properties")` -- `[]interface{}{}` in go syntax. I was trying to distinguish between the two configs with this, so it didn't work.

Also, there was another really strange issue when you had a non-mongodb repo: the error message only showed up on the first `terraform apply`. If you entered `terraform apply` again, you would be able to create a replica set sqlserver repo, for example. However, just moving the line 

```go
		properties = new(RepositoryProperties)
```

down into the `for` seemed to have fixed the issue (see latest commit).

Anyway, I just committed the changes. Thanks!